src/app/public/modules/restricted-view/restricted-view.directive.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  Directive,
+  TemplateRef,
+  ViewContainerRef,
+  Renderer2
+} from '@angular/core';
+
+import {
+  RestrictedViewAuthService
+} from './auth.service';
+
+@Directive({
+  selector: '[skyuxRestrictedView]'
+})
+export class RestrictedViewDirective {
+  constructor(
+    private authService: RestrictedViewAuthService,
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private renderer: Renderer2
+  ) {
+    this.viewContainer.clear();
+    this.authService.isAuthenticated.subscribe(isAuthenticated => {
```

Comment:
Subscriptions should be unsubscribed in `ngOnDestroy`. Here's an example using `ngUnsubscribe`: https://github.com/blackbaud/skyux-indicators/blob/master/src/app/public/modules/tokens/tokens.component.ts#L159-L161